In [1]:
import pandas as pd

In [4]:
M1 = pd.read_csv("/Users/swetarai/BLOOD_SYNOVIUM_COMBINE/gse_272_blood/annotated_exp_272.csv")
M1["Gene Symbol"].isnull().sum()
df_1 = M1[M1["Gene Symbol"].notnull()]
filtered_df_1 = df_1.loc[:, ['Gene Symbol'] + [col for col in df_1.columns if col.startswith("GSM")]]
filtered_df_1 = filtered_df_1.drop_duplicates(subset='Gene Symbol', keep='first')
filtered_df_1

Gene Symbol  GSM2449608  \
0                                       DDR1 /// MIR4640    2.928353   
1                                                   RFC2    2.681279   
2                                                  HSPA6    3.200373   
3                                                   PAX8    3.008607   
4                                                 GUCA1A    1.814431   
...                                                  ...         ...   
54531                         LOC102725263 /// LOC389906    2.381660   
54549                           C1orf50 /// LOC100129924    2.856775   
54577  FAM86B1 /// FAM86B2 /// FAM86C1 /// FAM86DP //...    2.488347   
54585                        HNRNPUL2 /// HNRNPUL2-BSCL2    2.308445   
54595                                       LOC100505915    2.783823   

       GSM2449609  GSM2449610  GSM2449611  GSM2449612  GSM2449613  GSM2449614  \
0        2.919860    2.862128    2.889364    2.920884    2.870000    2.927075   
1        2.642563    2.689835    2.705528    2.733763    2.726157    2.636703   
2        3.187529    3.253123    3.333800    3.266168    3.275608    3.190540   
3        2.991917    2.971649    3.020300    2.980154    2.990279    2.971847   
4        1.766750    1.773704    1.750219    1.768860    1.678898    1.837825   
...           ...         ...         ...         ...         ...         ...   
54531    2.393146    2.395874    2.546893    2.604183    2.430608    2.495660   
54549    2.840451    2.897739    2.901581    2.798850    2.858856    2.841344   
54577    2.581441    2.512073    2.534539    2.406974    2.433588    2.434192   
54585    2.254894    2.310908    2.444284    2.286731    2.212956    2.305959   
54595    2.771858    2.858048    2.666812    2.755194    2.722511    2.832467   

       GSM2449615  GSM2449616  ...  GSM2449878  GSM2449879  GSM2449880  \
0        2.881789    2.887258  ...    2.822652    2.876538    2.922520   
1        2.667755    2.666491  ...    2.748898    2.791219    2.736606   
2        3.253989    3.256648  ...    3.309197    3.285450    3.292569   
3        2.996316    2.954876  ...    2.928360    2.917507    2.989750   
4        1.840322    1.772330  ...    1.771276    1.772564    1.794683   
...           ...         ...  ...         ...         ...         ...   
54531    2.340232    2.542215  ...    2.458947    2.454049    2.416941   
54549    2.839988    2.848193  ...    2.897388    2.888492    2.898321   
54577    2.503806    2.554246  ...    2.288934    2.490519    2.526051   
54585    2.390426    2.268499  ...    2.239119    2.282043    2.358177   
54595    2.746938    2.639063  ...    2.690587    2.685683    2.627129   

       GSM2449881  GSM2449882  GSM2449883  GSM2449884  GSM2449885  GSM2449886  \
0        2.961124    2.924681    2.875892    2.854114    2.859889    2.891679   
1        2.608420    2.762922    2.808672    2.849820    2.751650    2.772160   
2        3.254789    3.343375    3.407475    3.324916    3.285933    3.272570   
3        3.015487    2.948581    2.970809    2.959193    2.939934    2.974253   
4        1.856459    1.785624    1.817391    1.711687    1.884332    1.772243   
...           ...         ...         ...         ...         ...         ...   
54531    2.545925    2.382622    2.349578    2.371773    2.440125    2.506577   
54549    2.919116    2.882134    2.824627    2.886832    2.869714    2.868541   
54577    2.584872    2.508250    2.303449    2.466768    2.426856    2.427709   
54585    2.193675    2.252132    2.157770    2.099963    2.373279    2.251390   
54595    2.725890    2.683343    2.644654    2.700746    2.908929    2.609138   

       GSM2449887  
0        2.901720  
1        2.718737  
2        3.270386  
3        2.949255  
4        1.818302  
...           ...  
54531    2.524236  
54549    2.876799  
54577    2.538408  
54585    2.333465  
54595    2.750716  

[23520 rows x 276 columns]

In [5]:
M2 = pd.read_csv("/Users/swetarai/BLOOD_SYNOVIUM_COMBINE/gse_776_blood/annotated_exp_776.csv")
M2["Gene Symbol"].isnull().sum()
df_2 = M2[M2["Gene Symbol"].notnull()]
filtered_df_2 = df_2.loc[:, ['Gene Symbol'] + [col for col in df_2.columns if col.startswith("GSM")]]
filtered_df_2 = filtered_df_2.drop_duplicates(subset='Gene Symbol', keep='first')
filtered_df_2

Gene Symbol  GSM2461861  GSM2461862  GSM2461863  \
0                 DDR1 /// MIR4640    2.919800    2.718915    2.885958   
1                             RFC2    2.711755    2.740913    2.664036   
2                            HSPA6    2.827558    3.029129    2.648849   
3                             PAX8    2.692826    2.640448    2.742146   
4                           GUCA1A    1.752539    1.705987    1.731654   
...                            ...         ...         ...         ...   
43437          MROH7-TTC4 /// TTC4    2.964383    2.922265    3.090700   
43530                       ZNF764    2.566866    2.553495    2.596264   
43532              LIME1 /// ZGPAT    3.043775    3.085888    2.987278   
43559     C1orf50 /// LOC100129924    2.758845    2.817083    2.808436   
43594  HNRNPUL2 /// HNRNPUL2-BSCL2    3.106832    3.104891    2.858177   

       GSM2461864  GSM2461865  GSM2461866  GSM2461867  GSM2461868  GSM2461869  \
0        2.740307    2.846462    2.719782    2.887556    2.728350    2.882078   
1        2.668755    2.763933    2.741677    2.687283    2.596055    2.784325   
2        3.024511    2.856237    2.938164    2.707857    3.049433    2.588101   
3        2.714944    2.669900    2.694252    2.768109    2.763225    2.719364   
4        1.756267    1.748544    1.946073    1.811534    1.726927    1.799126   
...           ...         ...         ...         ...         ...         ...   
43437    2.925558    3.022830    3.006738    2.916031    2.830859    2.989617   
43530    2.533309    2.690284    2.489422    2.586227    2.357394    2.600821   
43532    3.070985    3.041754    3.141697    3.042829    3.080050    3.089836   
43559    2.700590    2.748093    2.783237    2.725282    2.755404    2.737505   
43594    3.117686    3.037048    3.066090    3.068645    3.074041    3.083533   

       ...  GSM2462024  GSM2462025  GSM2462026  GSM2462027  GSM2462028  \
0      ...    2.496852    2.445417    2.605986    2.458237    2.436687   
1      ...    2.651780    2.674937    2.620808    2.542993    2.668176   
2      ...    2.689384    2.464534    2.661841    2.430124    2.517231   
3      ...    2.627737    2.623817    2.717520    2.646591    2.708606   
4      ...    1.817065    1.737537    1.749146    1.808464    1.762084   
...    ...         ...         ...         ...         ...         ...   
43437  ...    3.100604    3.049971    3.053502    3.116701    3.015466   
43530  ...    2.868476    2.807953    2.723576    2.838261    2.716166   
43532  ...    3.133409    3.069036    3.120576    3.136881    3.138159   
43559  ...    2.828772    2.852453    2.774096    2.824579    2.855371   
43594  ...    3.113504    3.075065    3.106204    3.007433    3.087815   

       GSM2462029  GSM2462030  GSM2462031  GSM2462032  GSM2462033  
0        2.418759    2.463751    2.360704    2.385554    2.431343  
1        2.544346    2.590074    2.523214    2.671206    2.564830  
2        2.719061    2.777088    2.366575    2.763958    2.494746  
3        2.710897    2.558635    2.607487    2.549335    2.557658  
4        1.794041    1.749932    1.745319    1.824347    1.748276  
...           ...         ...         ...         ...         ...  
43437    3.055863    3.114304    3.042095    2.996378    3.044077  
43530    2.780098    2.672861    2.732664    2.706315    2.875363  
43532    3.093241    3.067253    3.043156    3.109793    3.183514  
43559    2.764181    2.827804    2.872246    2.818023    2.757565  
43594    3.103394    3.118542    3.054452    3.162893    3.092004  

[21337 rows x 174 columns]

In [6]:
merged_data = pd.merge(filtered_df_1, filtered_df_2, on="Gene Symbol", how ='inner')
merged_data = merged_data.rename(columns={'Gene Symbol': 'Gene_Symbol'})
merged_data

Gene_Symbol  GSM2449608  GSM2449609  GSM2449610  \
0                 DDR1 /// MIR4640    2.928353    2.919860    2.862128   
1                             RFC2    2.681279    2.642563    2.689835   
2                            HSPA6    3.200373    3.187529    3.253123   
3                             PAX8    3.008607    2.991917    2.971649   
4                           GUCA1A    1.814431    1.766750    1.773704   
...                            ...         ...         ...         ...   
21332         MINOS1-NBL1 /// NBL1    2.902807    2.862354    2.909786   
21333          MROH7-TTC4 /// TTC4    2.950347    2.954725    2.966098   
21334              LIME1 /// ZGPAT    2.959096    2.957979    3.007961   
21335     C1orf50 /// LOC100129924    2.856775    2.840451    2.897739   
21336  HNRNPUL2 /// HNRNPUL2-BSCL2    2.308445    2.254894    2.310908   

       GSM2449611  GSM2449612  GSM2449613  GSM2449614  GSM2449615  GSM2449616  \
0        2.889364    2.920884    2.870000    2.927075    2.881789    2.887258   
1        2.705528    2.733763    2.726157    2.636703    2.667755    2.666491   
2        3.333800    3.266168    3.275608    3.190540    3.253989    3.256648   
3        3.020300    2.980154    2.990279    2.971847    2.996316    2.954876   
4        1.750219    1.768860    1.678898    1.837825    1.840322    1.772330   
...           ...         ...         ...         ...         ...         ...   
21332    2.861810    2.897746    2.830013    2.864319    2.854887    2.841305   
21333    2.958894    2.940008    2.891876    2.906382    2.904471    2.949384   
21334    3.012335    3.028611    2.991183    3.003196    3.035986    3.029550   
21335    2.901581    2.798850    2.858856    2.841344    2.839988    2.848193   
21336    2.444284    2.286731    2.212956    2.305959    2.390426    2.268499   

       ...  GSM2462024  GSM2462025  GSM2462026  GSM2462027  GSM2462028  \
0      ...    2.496852    2.445417    2.605986    2.458237    2.436687   
1      ...    2.651780    2.674937    2.620808    2.542993    2.668176   
2      ...    2.689384    2.464534    2.661841    2.430124    2.517231   
3      ...    2.627737    2.623817    2.717520    2.646591    2.708606   
4      ...    1.817065    1.737537    1.749146    1.808464    1.762084   
...    ...         ...         ...         ...         ...         ...   
21332  ...    2.432038    2.517036    2.394108    2.749605    2.392589   
21333  ...    3.100604    3.049971    3.053502    3.116701    3.015466   
21334  ...    3.133409    3.069036    3.120576    3.136881    3.138159   
21335  ...    2.828772    2.852453    2.774096    2.824579    2.855371   
21336  ...    3.113504    3.075065    3.106204    3.007433    3.087815   

       GSM2462029  GSM2462030  GSM2462031  GSM2462032  GSM2462033  
0        2.418759    2.463751    2.360704    2.385554    2.431343  
1        2.544346    2.590074    2.523214    2.671206    2.564830  
2        2.719061    2.777088    2.366575    2.763958    2.494746  
3        2.710897    2.558635    2.607487    2.549335    2.557658  
4        1.794041    1.749932    1.745319    1.824347    1.748276  
...           ...         ...         ...         ...         ...  
21332    2.297055    2.341876    2.503396    2.242118    2.471449  
21333    3.055863    3.114304    3.042095    2.996378    3.044077  
21334    3.093241    3.067253    3.043156    3.109793    3.183514  
21335    2.764181    2.827804    2.872246    2.818023    2.757565  
21336    3.103394    3.118542    3.054452    3.162893    3.092004  

[21337 rows x 449 columns]

In [7]:
merged_data= merged_data.set_index("Gene_Symbol").T

In [8]:
merged_data = merged_data.reset_index()
print(merged_data.columns)
merged_data = merged_data.rename(columns={'index': 'Gene_Symbol'})

Index(['index', 'DDR1 /// MIR4640', 'RFC2', 'HSPA6', 'PAX8', 'GUCA1A',
       'MIR5193 /// UBA7', 'THRA', 'PTPN21', 'CCL5',
       ...
       'RP11-495P10.6', 'LOC101929475', 'RP11-1260E13.2', 'LOC389641', 'GALR3',
       'MINOS1-NBL1 /// NBL1', 'MROH7-TTC4 /// TTC4', 'LIME1 /// ZGPAT',
       'C1orf50 /// LOC100129924', 'HNRNPUL2 /// HNRNPUL2-BSCL2'],
      dtype='object', name='Gene_Symbol', length=21338)


In [9]:
merged_data

Gene_Symbol Gene_Symbol  DDR1 /// MIR4640      RFC2     HSPA6      PAX8  \
0            GSM2449608          2.928353  2.681279  3.200373  3.008607   
1            GSM2449609          2.919860  2.642563  3.187529  2.991917   
2            GSM2449610          2.862128  2.689835  3.253123  2.971649   
3            GSM2449611          2.889364  2.705528  3.333800  3.020300   
4            GSM2449612          2.920884  2.733763  3.266168  2.980154   
..                  ...               ...       ...       ...       ...   
443          GSM2462029          2.418759  2.544346  2.719061  2.710897   
444          GSM2462030          2.463751  2.590074  2.777088  2.558635   
445          GSM2462031          2.360704  2.523214  2.366575  2.607487   
446          GSM2462032          2.385554  2.671206  2.763958  2.549335   
447          GSM2462033          2.431343  2.564830  2.494746  2.557658   

Gene_Symbol    GUCA1A  MIR5193 /// UBA7      THRA    PTPN21      CCL5  ...  \
0            1.814431          3.035352  2.794960  1.993711  3.573319  ...   
1            1.766750          3.072702  2.880967  2.036722  3.561053  ...   
2            1.773704          3.045336  2.872246  1.892788  3.585422  ...   
3            1.750219          3.086978  2.900855  2.033047  3.598813  ...   
4            1.768860          3.085212  2.806505  1.898818  3.527495  ...   
..                ...               ...       ...       ...       ...  ...   
443          1.794041          3.221841  2.933400  2.110611  3.539862  ...   
444          1.749932          3.268715  2.886809  1.995671  3.566206  ...   
445          1.745319          3.198502  2.913508  2.104843  3.526773  ...   
446          1.824347          3.238655  2.987705  1.898630  3.551848  ...   
447          1.748276          3.268024  2.981201  2.103751  3.538103  ...   

Gene_Symbol  RP11-495P10.6  LOC101929475  RP11-1260E13.2  LOC389641     GALR3  \
0                 2.062285      2.220692        1.929662   2.531248  2.768501   
1                 2.170677      2.220664        1.706599   2.541923  2.725094   
2                 2.236451      2.302767        1.861940   2.497606  2.771164   
3                 2.291486      2.268351        1.880827   2.453638  2.732430   
4                 2.202777      2.183851        1.870881   2.527991  2.750591   
..                     ...           ...             ...        ...       ...   
443               2.381043      2.425481        1.808563   2.541818  2.321079   
444               2.372905      2.421513        1.723371   2.478416  2.244271   
445               2.242343      2.268841        1.747862   2.340760  2.292508   
446               2.283267      2.200476        1.765177   2.434111  2.176570   
447               2.290919      2.169370        1.753216   2.626630  2.280043   

Gene_Symbol  MINOS1-NBL1 /// NBL1  MROH7-TTC4 /// TTC4  LIME1 /// ZGPAT  \
0                        2.902807             2.950347         2.959096   
1                        2.862354             2.954725         2.957979   
2                        2.909786             2.966098         3.007961   
3                        2.861810             2.958894         3.012335   
4                        2.897746             2.940008         3.028611   
..                            ...                  ...              ...   
443                      2.297055             3.055863         3.093241   
444                      2.341876             3.114304         3.067253   
445                      2.503396             3.042095         3.043156   
446                      2.242118             2.996378         3.109793   
447                      2.471449             3.044077         3.183514   

Gene_Symbol  C1orf50 /// LOC100129924  HNRNPUL2 /// HNRNPUL2-BSCL2  
0                            2.856775                     2.308445  
1                            2.840451                     2.254894  
2                            2.897739                     2.310908  
3           

# PHENODATA

In [10]:
phenodata1 = pd.read_csv("/Users/swetarai/BLOOD_SYNOVIUM_COMBINE/GPL570_2_DATA_BLOOD_SYNO/merged_pheno_2_dataset_GPL570.csv")
phenodata1

Gene_Symbol    disease_state sample_type  gender
0    GSM2449608  Healthy_control       Blood  Female
1    GSM2449609  Healthy_control       Blood  Female
2    GSM2449610  Healthy_control       Blood  Female
3    GSM2449611  Healthy_control       Blood  Female
4    GSM2449612  Healthy_control       Blood  Female
..          ...              ...         ...     ...
446  GSM2462029               RA       Blood  Female
447  GSM2462030               RA       Blood    Male
448  GSM2462031               RA       Blood  Female
449  GSM2462032               RA       Blood  Female
450  GSM2462033               RA       Blood  Female

[451 rows x 4 columns]

In [11]:
df_final_1 = pd.merge(merged_data, phenodata1[['Gene_Symbol', 'gender','sample_type','disease_state']], on='Gene_Symbol', how='left')
df_final_1

Gene_Symbol  DDR1 /// MIR4640      RFC2     HSPA6      PAX8    GUCA1A  \
0    GSM2449608          2.928353  2.681279  3.200373  3.008607  1.814431   
1    GSM2449609          2.919860  2.642563  3.187529  2.991917  1.766750   
2    GSM2449610          2.862128  2.689835  3.253123  2.971649  1.773704   
3    GSM2449611          2.889364  2.705528  3.333800  3.020300  1.750219   
4    GSM2449612          2.920884  2.733763  3.266168  2.980154  1.768860   
..          ...               ...       ...       ...       ...       ...   
443  GSM2462029          2.418759  2.544346  2.719061  2.710897  1.794041   
444  GSM2462030          2.463751  2.590074  2.777088  2.558635  1.749932   
445  GSM2462031          2.360704  2.523214  2.366575  2.607487  1.745319   
446  GSM2462032          2.385554  2.671206  2.763958  2.549335  1.824347   
447  GSM2462033          2.431343  2.564830  2.494746  2.557658  1.748276   

     MIR5193 /// UBA7      THRA    PTPN21      CCL5  ...  LOC389641     GALR3  \
0            3.035352  2.794960  1.993711  3.573319  ...   2.531248  2.768501   
1            3.072702  2.880967  2.036722  3.561053  ...   2.541923  2.725094   
2            3.045336  2.872246  1.892788  3.585422  ...   2.497606  2.771164   
3            3.086978  2.900855  2.033047  3.598813  ...   2.453638  2.732430   
4            3.085212  2.806505  1.898818  3.527495  ...   2.527991  2.750591   
..                ...       ...       ...       ...  ...        ...       ...   
443          3.221841  2.933400  2.110611  3.539862  ...   2.541818  2.321079   
444          3.268715  2.886809  1.995671  3.566206  ...   2.478416  2.244271   
445          3.198502  2.913508  2.104843  3.526773  ...   2.340760  2.292508   
446          3.238655  2.987705  1.898630  3.551848  ...   2.434111  2.176570   
447          3.268024  2.981201  2.103751  3.538103  ...   2.626630  2.280043   

     MINOS1-NBL1 /// NBL1  MROH7-TTC4 /// TTC4  LIME1 /// ZGPAT  \
0                2.902807             2.950347         2.959096   
1                2.862354             2.954725         2.957979   
2                2.909786             2.966098         3.007961   
3                2.861810             2.958894         3.012335   
4                2.897746             2.940008         3.028611   
..                    ...                  ...              ...   
443              2.297055             3.055863         3.093241   
444              2.341876             3.114304         3.067253   
445              2.503396             3.042095         3.043156   
446              2.242118             2.996378         3.109793   
447              2.471449             3.044077         3.183514   

     C1orf50 /// LOC100129924  HNRNPUL2 /// HNRNPUL2-BSCL2  gender  \
0                    2.856775                     2.308445  Female   
1                    2.840451                     2.254894  Female   
2                    2.897739                     2.310908  Female   
3                    2.901581                     2.444284  Female   
4                    2.798850                     2.286731  Female   
..                        ...                          ...     ...   
443                  2.764181                     3.103394  Female   
444                  2.827804                     3.118542    Male   
445                  2.872246                     3.054452  Female   
446                  2.818023                     3.162893  Female   
447                  2.757565                     3.092004  Female   

     sample_type    disease_state  
0          Blood  Healthy_control  
1          Blood  Healthy_control  
2          Blood  Healthy_control  
3          Blood  Healthy_control  
4          Blood  Healthy_control  
..           ...              ...  
443        Blood               RA  
444        Blood               RA  
445        Blood               RA  
446        Blood               RA  
447        Blood               RA  

[448 rows x 21341 columns]

# take this into consideration as training dataset

### it has 449 samples and 21341 probes or features for GPL570 both BLOOD and SYNOVIUM

In [13]:
df_final_1.to_csv("/Users/swetarai/BLOOD_SYNOVIUM_COMBINE/GPL570_2_DATA_BLOOD_SYNO/272_776_GPL570_BLOOD_SYNO_exp_pheno.csv")